storage/blob.go

```diff
@@ -561,13 +528,33 @@ func (b BlobStorageClient) GetContainerPermissions(container string, timeout int
 		return nil, err
 	}
 
-	permissionResponse = &ContainerAccessResponse{}
-	permissionResponse.AccessPolicy = out.SignedIdentifiersList
-	permissionResponse.ContainerAccess = ContainerAccessType(containerAccess)
+	permissionResponse = b.updateAccessPolicy(out, &resp.headers)
 
 	return permissionResponse, nil
 }
 
+func (b BlobStorageClient) updateAccessPolicy(ap AccessPolicy, headers *http.Header) *ContainerPermissions {
```

Comment:
I'm not sure why this function is tied on to `BlobStorageClient`. It seems like this just returns a ContainerPermissions with the appropriate `AccessPolicy` enforced. Maybe this should be another ContainerPermissions constructor as mentioned below.